jerry-core/vm/vm.c

```diff
@@ -2922,6 +2922,19 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
           ecma_fast_free_value (value);
           continue;
         }
+#if ENABLED (JERRY_ESNEXT)
+        case VM_OC_BRANCH_IF_NULLISH:
+        {
+          ecma_value_t value = *(--stack_top_p);
```

Comment:
I'd reverse the logic. Instead of pop and push back, just read the stack_top[-1] and only pop, if it is undefined or null.